go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -120,84 +114,102 @@ func (qre *QueryExecutor) Execute() (reply *sqltypes.Result, err error) {
-			}
-			return qre.txFetch(conn, qre.plan.FullQuery, qre.bindVars, nil, "", true, true)
-		case planbuilder.PlanInsertPK:
-			return qre.execInsertPK(conn)
-		case planbuilder.PlanInsertMessage:
-			return qre.execInsertMessage(conn)
-		case planbuilder.PlanInsertSubquery:
-			return qre.execInsertSubquery(conn)
-		case planbuilder.PlanDMLPK:
-			return qre.execDMLPK(conn)
-		case planbuilder.PlanDMLSubquery:
-			return qre.execDMLSubquery(conn)
-		case planbuilder.PlanOtherRead, planbuilder.PlanOtherAdmin:
-			return qre.execSQL(conn, qre.query, true)
-		case planbuilder.PlanUpsertPK:
-			return qre.execUpsertPK(conn)
-		case planbuilder.PlanSet:
-			return qre.txFetch(conn, qre.plan.FullQuery, qre.bindVars, nil, "", true, true)
-		case planbuilder.PlanPassSelect, planbuilder.PlanSelectLock, planbuilder.PlanSelectImpossible:
-			return qre.execDirect(conn)
-		default:
-			// handled above:
-			// planbuilder.PlanNextval
-			// planbuilder.PlanDDL
-
-			// not valid for Execute:
-			// planbuilder.PlanSelectStream
-			// planbuilder.PlanMessageStream:
-			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "%s unexpected plan type", qre.plan.PlanID.String())
+		return qre.txConnExec(conn)
+	}
+
+	switch qre.plan.PlanID {
+	case planbuilder.PlanSelect, planbuilder.PlanSelectImpossible:
+		maxrows := qre.getSelectLimit()
+		qre.bindVars["#maxLimit"] = sqltypes.Int64BindVariable(maxrows + 1)
+		qr, err := qre.execSelect()
+		if err != nil {
+			return nil, err
 		}
-	} else {
-		switch qre.plan.PlanID {
-		case planbuilder.PlanPassSelect, planbuilder.PlanSelectImpossible:
-			return qre.execSelect()
-		case planbuilder.PlanSelectLock:
-			return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "%s disallowed outside transaction", qre.plan.PlanID.String())
-		case planbuilder.PlanSet:
-			return qre.execSet()
-		case planbuilder.PlanOtherRead:
-			conn, connErr := qre.getConn()
-			if connErr != nil {
-				return nil, connErr
-			}
-			defer conn.Recycle()
-			return qre.execSQL(conn, qre.query, true)
-
-		case planbuilder.PlanPassDML:
-			fallthrough
-		case planbuilder.PlanInsertPK:
-			fallthrough
-		case planbuilder.PlanInsertMessage:
-			fallthrough
-		case planbuilder.PlanInsertSubquery:
-			fallthrough
-		case planbuilder.PlanDMLPK:
-			fallthrough
-		case planbuilder.PlanDMLSubquery:
-			fallthrough
-		case planbuilder.PlanUpsertPK:
-			if !qre.tsv.qe.autoCommit.Get() {
-				return nil, vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "%s disallowed outside transaction", qre.plan.PlanID.String())
-			}
-			return qre.execDmlAutoCommit()
-		default:
-			// handled above:
-			// planbuilder.PlanNextval
-			// planbuilder.PlanDDL
-
-			// not valid for Execute:
-			// planbuilder.PlanSelectStream
-			// planbuilder.PlanMessageStream:
-			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "%s unexpected plan type", qre.plan.PlanID.String())
+		if err := qre.verifyRowCount(int64(len(qr.Rows)), maxrows); err != nil {
+			return nil, err
```

Comment:
Test exists for DML Limit.  Not able to find a test for select limit.

Reply:
Good catch. I must have accidentally deleted an existing test.